target/linux/lantiq/patches-5.10/0801-remoteproc-add-avm_wasp-driver-kernel-module.patch

```diff
@@ -0,0 +1,1381 @@
++				goto err;
++			}
++		}
++	}
++
++	ret = of_property_read_u32(dev->of_node, "wasp-netboot-mdio",
++				   &phandle);
++	if (ret) {
++		dev_err(dev, "No wasp-netboot-mdio given\n");
++		goto err;
++	} else {
++		struct device_node *mdio_node =
++					of_find_node_by_phandle(phandle);
++
++		if (!mdio_node) {
++			dev_err(dev, "Get wasp-netboot-mdio failed\n");
++			ret = -ENODEV;
++			goto err;
++		} else {
++			avmwasp->mdio_bus = of_mdio_find_bus(mdio_node);
++			of_node_put(mdio_node);
++			if (!avmwasp->mdio_bus) {
++				dev_err(dev, "mdio bus not found\n");
++				ret = -ENODEV;
++				goto err;
++			}
++			avmwasp->mdio_bus_id = avmwasp->mdio_bus->id;
++			put_device(&avmwasp->mdio_bus->dev);
++		}
++	}
++
++	avmwasp->startup_gpio = of_get_named_gpio_flags(dev->of_node,
++							"startup-gpio",
++							0,
++							&avmwasp->s_gpio_flg);
++	if (!gpio_is_valid(avmwasp->startup_gpio)) {
++		dev_err(dev, "Request wasp-startup gpio failed\n");
++		ret = -ENODEV;
++		goto err;
++	} else {
++		ret = devm_gpio_request_one(dev, avmwasp->startup_gpio,
++					    (avmwasp->s_gpio_flg &
++					    OF_GPIO_ACTIVE_LOW) ?
++					    GPIOF_OUT_INIT_LOW :
++					    GPIOF_OUT_INIT_HIGH,
++					    "wasp-startup");
++
++		if (ret) {
++			dev_err(dev, "get wasp-startup gpio failed\n");
++			goto err;
++		}
++	}
++
++	avmwasp->reset_gpio = of_get_named_gpio_flags(dev->of_node,
++						      "reset-gpio",
++						      0,
++						      &avmwasp->r_gpio_flg);
++	if (!gpio_is_valid(avmwasp->reset_gpio)) {
++		dev_err(dev, "Request wasp-reset gpio failed\n");
++		ret = -ENODEV;
++		goto err_free_startup_gpio;
++	} else {
++		ret = devm_gpio_request_one(dev, avmwasp->reset_gpio,
++					    (avmwasp->r_gpio_flg &
++					    OF_GPIO_ACTIVE_LOW)	?
++					    GPIOF_OUT_INIT_LOW :
++					    GPIOF_OUT_INIT_HIGH,
++					    "wasp-reset");
++
++		if (ret) {
++			dev_err(dev, "get wasp-reset gpio failed\n");
++			goto err_free_startup_gpio;
++		}
++	}
++
++	ret = of_property_read_string(dev->of_node, "wasp-netboot-firmware",
++				      (const char **)
++				      &avmwasp->netboot_firmware);
++	if (ret) {
++		dev_err(dev, "No WASP network boot firmware filename given\n");
++		goto err_free_reset_gpio;
++	}
++
++	ret = request_firmware_direct((const struct firmware **)
++			&avmwasp->eeprom_blob, avmwasp->netboot_firmware, dev);
++	if (ret) {
++		dev_err(dev, "Could not load WASP network boot firmware\n");
++		goto err_free_reset_gpio;
++	}
++
++	if (avmwasp->eeprom_blob->size > 0xffff) {
++		dev_err(dev, "WASP network boot firmware too big\n");
++		ret = -EINVAL;
++		goto err_free_reset_gpio;
++	}
++
++	avm_wasp_firmware_release(avmwasp);
++
++	dev_set_drvdata(dev, rproc);
++
++	ret = devm_rproc_add(dev, rproc);
++	if (ret) {
++		dev_err(dev, "rproc_add failed\n");
++		goto err_free_reset_gpio;
++	}
++
++	return 0;
++
++err_free_reset_gpio:
++	devm_gpio_free(&avmwasp->pdev->dev, avmwasp->reset_gpio);
++	gpio_set_value(avmwasp->startup_gpio,
```

Comment:
gpio_ is being phased out in favor or gpiod_

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=79a9becda8940deb2274b5aa4577c86d52ee7ecb

Reply:
Its removed too in v2.